### ISSUE TITLE
initial install with letsencrypt fails

### ISSUE THREAD

**Issue by Placinta** [liaralabs/swizzin#81]
Ubuntu 16.04, initial install of swizzin, chose rtorrent, rutorrent, letsencrypt, nginx.
Downloaded the setup script with wget, and then executed sudo setup.sh.

I get this output:

```
....
Installing letsencrypt
Enter domain name to secure with LE
subdomain.maindomain.com (< ---- this is obviously a dummy value, but I entered a real, valid domain that i own)
Do you want to apply this certificate to your swizzin default conf? (y/n) y
Is your DNS managed by CloudFlare? (y/n) n
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   705  100   705    0     0    479      0  0:00:01  0:00:01 --:--:--   479
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  163k  100  163k    0     0   592k      0 --:--:-- --:--:-- --:--:--  593k
[Sat Mar 31 18:07:24 CEST 2018] Installing from online archive.
[Sat Mar 31 18:07:24 CEST 2018] Downloading https://github.com/Neilpang/acme.sh/archive/master.tar.gz
[Sat Mar 31 18:07:25 CEST 2018] Extracting master.tar.gz
[Sat Mar 31 18:07:25 CEST 2018] Installing to /home/placinta/.acme.sh
[Sat Mar 31 18:07:25 CEST 2018] Installed to /home/placinta/.acme.sh/acme.sh
[Sat Mar 31 18:07:25 CEST 2018] Installing alias to '/home/placinta/.bashrc'
[Sat Mar 31 18:07:25 CEST 2018] OK, Close and reopen your terminal to start using acme.sh
[Sat Mar 31 18:07:26 CEST 2018] Installing cron job
no crontab for root
no crontab for root
[Sat Mar 31 18:07:26 CEST 2018] Good, bash is found, so change the shebang to use bash as preferred.
[Sat Mar 31 18:07:26 CEST 2018] OK
[Sat Mar 31 18:07:26 CEST 2018] Install success!
/usr/local/bin/swizzin/install/letsencrypt.sh: line 109: /root/.acme.sh/acme.sh: No such file or directory
ERROR: Certificate could not be issued. Please check your info and try again
Package install took 13 minutes and 50 seconds
Installation complete!
```

I noticed that acme.sh is installed to my current user home folder, and not into /root or the new user I specify via setup.sh.

**Comment by liaralabs** [comment]
This is an unfortunate side effect of the way defaults are set for sudo under Ubuntu. By default, Ubuntu assumes that you should not inherit the home directory of the user you sudo **to**, rather keep the home directory of the user you sudo **from**.

While I don't currently address this issue specifically for Ubuntu users, there's also nothing I can do to change defaults for users who are running the script as sudo for the first time. In this case, I recommend running the setup.sh script with `sudo -H bash setup.sh` as this will override the default and set it to the user you are sudoing to.

However, that said, I do try to issue absolute paths wherever possible to avoid this issue in the first place. I'll add a check in the let's encrypt installer to check if the home variable reads as root and adjust it if necessary.

**Comment by Placinta** [comment]
So yeah, it seems that .acme.sh was copied to my main user at /home/placinta, whereas the script execution expects it at /root/. If i copy over the files, and rerun letsencrypt installation, everything worked out.

But that raises another question for me. Is there any support at all to run box not logged in as root?

**Comment by liaralabs** [comment]
> But that raises another question for me. Is there any support at all to run box not logged in as root?

No, if you'll notice, `box` isn't in your path as a regular user. Just about every command in the box script requires root in some form or another. You either need to sudo or su - before you can even access the script.

**Comment by Placinta** [comment]
But as I mentioned above, the only choice I have is to run sudo with absolute path to box. Because I disabled root login.
I'm sure you know that as a security measure, everyone recommends disabling root login.

**Comment by liaralabs** [comment]
> Even if I add the path manually to my .bashrc / .zshrc, which box still doesn't find it, because /usr/local/bin/swizzin is a symlink to /etc/swizzin/scripts/ , whic has no permission for reading / executing except root.

This is 100% intentional. I do not want people to think they can run scripts in this directory without root, because, in short, they pretty much shouldn't run any of them without root.

>Would it not be better to somehow adjust the installation routine, to allow no-root-login people like me to use sudo box out-of-the box after installation?

I am going to add a check for ubuntu and modify the defaults accordingly; however as I stated I cannot modify the defaults **before** you run the script. That's what `sudo -H` is for -- you seem to have glossed over this bit of info in my previous post.

Sudo will work the same as su, but for Ubuntu you need to specify the `-H` argument ***by default***.

I'll add a check in the updater, which will fix it for those who have installed/are installing swizzin, but you will still need to specify -H if that's how you're choosing to run the initial install.

**Comment by Placinta** [comment]
Ok. I believe I understand now. I was thinking maybe create some wrapper scripts that would be accessible to any user, but would then execute sudo 'appropriate_script', but I don't know how feasible that is.

Thank you for your help, and have a great day.

**Comment by liaralabs** [comment]
The issue remains the same either way -- sudo inherits a set of environmental variables by default. These are typically built at compile time for sudo. Debian uses different defaults than Ubuntu, which is the cause of this discrepency

**Comment by liaralabs** [comment]
This should be fixed now in bd6752e4af52eebefbd704f549b3620d3ef64b75
